### PR TITLE
Package readme's for NuGet.VisualStudio and NuGet.VisualStudio.Contracts

### PR DESCRIPTION
**Pull request by martinrrm**
## Bug

Fixes:
https://github.com/NuGet/Client.Engineering/issues/1507 (this is just one package and I dont want to close the issue)
Regression? Last working version:

## Description

Adding package README's for NuGet.VisualStudio and NuGet.VisualStudio.Contracts
## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Review by JonDouglas — DISMISSED** [review]
Great!

File: src/NuGet.Clients/NuGet.VisualStudio/README.md [hunk 1]
```diff
@@ -1 +1,35 @@
...
+public IVsPackageInstaller2 packageInstaller;
+packageInstaller.InstallLatestPackage(null, currentProject,
+    "Newtonsoft.Json", false, false);
+
+//Using the IComponentModel service
+var componentModel = (IComponentModel)GetService(typeof(SComponentModel));
+IVsPackageInstallerServices installerServices =
+    componentModel.GetService<IVsPackageInstallerServices>();
+
+var installedPackages = installerServices.GetInstalledPackages();
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.VisualStudio/README.md]: We should not use this example as it's deprecated.

**Review by jeffkl — DISMISSED** [review]

File: src/NuGet.Clients/NuGet.VisualStudio.Contracts/README.md [hunk 2]
```diff
@@ -1 +1,33 @@
...
+
+Install the NuGet.VisualStudio.Contracts package into your project, as well as [Microsoft.VisualStudio.SDK](https://www.nuget.org/packages/Microsoft.VisualStudio.SDK).
+
+Use the IAsyncServiceProvider to get Visual Studio's service broker, and use that to get NuGet's service. Note that [AsyncPackage extends IVsAsyncServiceProvider2](https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.asyncpackage?view=visualstudiosdk-2022), so your class that implements AsyncPackage can be used as the IAsyncServiceProvider. Also see the docs on [IBrokeredServiceContainer](https://learn.microsoft.com/dotnet/api/microsoft.visualstudio.shell.servicebroker.ibrokeredservicecontainer?view=visualstudiosdk-2022) and [IServiceBroker](https://learn.microsoft.com/dotnet/api/microsoft.servicehub.framework.iservicebroker?view=visualstudiosdk-2022)
+
+```c#
+// Your AsyncPackage implements IAsyncServiceProvider
+IAsyncServiceProvider asyncServiceProvider = this;
+var brokeredServiceContainer = await asyncServiceProvider.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>();
+var serviceBroker = brokeredServiceContainer.GetFullAccessServiceBroker();
```
Comment (erdembayar) [inline src/NuGet.Clients/NuGet.VisualStudio.Contracts/README.md]:
```suggestion
IServiceBroker serviceBroker = brokeredServiceContainer.GetFullAccessServiceBroker();
```
This way it's clearer that we're using `IServiceBroker`

File: src/NuGet.Clients/NuGet.VisualStudio.Contracts/README.md [hunk 3]
```diff
@@ -1 +1,33 @@
...
+var nugetProjectService = await serviceBroker.GetProxyAsync<INuGetProjectService>(NuGetServices.NuGetProjectServiceV1);
```
Comment (erdembayar) [inline src/NuGet.Clients/NuGet.VisualStudio.Contracts/README.md]: Probably same here, add explicit type.

File: src/NuGet.Clients/NuGet.VisualStudio.Contracts/README.md [hunk 4]
```diff
@@ -1 +1,33 @@
...
+
+## Example
+
+### Get installed packages of a project.
+
+```c#
+InstalledPackagesResult installedPackagesResult;
+using (nugetProjectService as IDisposable)
+{
+    installedPackagesResult = await nugetProjectService.GetInstalledPackages(projectGuid, cancellationToken);
```
Comment (erdembayar) [inline src/NuGet.Clients/NuGet.VisualStudio.Contracts/README.md]: @nkolev92 
Does this one need to wait for nomination finish before calling this?

File: src/NuGet.Clients/NuGet.VisualStudio/README.md [hunk 5]
```diff
@@ -1 +1,35 @@
-APIs for invoking NuGet services in Visual Studio.
+# NuGet.VisualStudio.Contracts
```
Comment (erdembayar) [inline src/NuGet.Clients/NuGet.VisualStudio/README.md]:
```suggestion
# NuGet.VisualStudio
```

File: src/NuGet.Clients/NuGet.VisualStudio.Contracts/README.md (hunk 3 above)
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.VisualStudio.Contracts/README.md]: This is one of those  where I'd argue the type is obvious, the whole `GetProxyAsync<INuGetProjectService>`. but I also don't mind changing it as it's for a readme.

File: src/NuGet.Clients/NuGet.VisualStudio/README.md (hunk 1 above)
Comment (erdembayar) [reply src/NuGet.Clients/NuGet.VisualStudio/README.md]: It looks like new one is:
Task<InstalledPackagesResult> GetInstalledPackagesAsync(Guid projectId, CancellationToken cancellationToken)

File: src/NuGet.Clients/NuGet.VisualStudio.Contracts/README.md (hunk 4 above)
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.VisualStudio.Contracts/README.md]: This actually isn't even a valid call. 
This is an async method that doesn't exist.

File: src/NuGet.Clients/NuGet.VisualStudio.Contracts/README.md [hunk 6]
```diff
@@ -1 +1,33 @@
...
+}
+```
+
+## Aditional documentation
+
+More information about the NuGet.Protocol library can be found on the [official Microsoft documentation page](https://learn.microsoft.com/nuget/reference/nuget-client-sdk#nugetprotocol) and [NuGet API docs](https://learn.microsoft.com/nuget/visual-studio-extensibility/nuget-api-in-visual-studio).
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.VisualStudio.Contracts/README.md]: We don't need to mention NuGet.Protocol in this PR.

Comment (zivkan) [reply src/NuGet.Clients/NuGet.VisualStudio.Contracts/README.md]: In fact, in Visual Studio, we don't provide binding redirects for NuGet.Protocol, or any of the other "NuGet Client SDK" assemblies. Therefore, we don't officially support using any of those packages in a VS extension. We only support NuGet.VisualStudio and NuGet.VisualStudio.Contracts.

File: src/NuGet.Clients/NuGet.VisualStudio.Contracts/README.md [hunk 7]
```diff
@@ -1 +1,33 @@
...
+
+```c#
+InstalledPackagesResult installedPackagesResult;
+using (nugetProjectService as IDisposable)
+{
+    installedPackagesResult = await nugetProjectService.GetInstalledPackagesAsync(projectGuid, cancellationToken);
+}
+```
+
+## Aditional documentation
```
Comment (jeffkl) [inline src/NuGet.Clients/NuGet.VisualStudio.Contracts/README.md]:
```suggestion
## Additional documentation
```

File: src/NuGet.Clients/NuGet.VisualStudio/README.md [hunk 8]
```diff
@@ -1 +1,35 @@
...
+From NuGet 6.0, all of these APIs are available in the package [NuGet.VisualStudio](https://www.nuget.org/packages/NuGet.VisualStudio/). In NuGet 5.11 and earlier, the APIs in the namespace [`NuGet.VisualStudio`](https://learn.microsoft.com/dotnet/api/microsoft.visualstudio.componentmodelhost.icomponentmodel?view=visualstudiosdk-2022) are available in the package [NuGet.VisualStudio](https://www.nuget.org/packages/NuGet.VisualStudio/), and APIs in the namespace [`NuGet.SolutionRestoreManager`](https://learn.microsoft.com/dotnet/api/microsoft.visualstudio.componentmodelhost.icomponentmodel?view=visualstudiosdk-2022) are available in the package [NuGet.SolutionRestoreManager.Interop](https://www.nuget.org/packages/NuGet.SolutionRestoreManager.Interop/).
+
+## Example
+
+### Get installed packages
+
+To use a service, import it through the [MEF Import attribute](https://learn.microsoft.com/dotnet/framework/mef/#imports-and-exports-with-attributes), or through the [IComponentModel service](https://learn.microsoft.com/dotnet/api/microsoft.visualstudio.componentmodelhost.icomponentmodel?view=visualstudiosdk-2022).
+
+```c#
+//Using the Import attribute
```
Comment (jeffkl) [inline src/NuGet.Clients/NuGet.VisualStudio/README.md]:
```suggestion
// Using the Import attribute
```

File: src/NuGet.Clients/NuGet.VisualStudio/README.md [hunk 9]
```diff
@@ -1 +1,35 @@
...
+
+//Using the IComponentModel service
+var componentModel = (IComponentModel)GetService(typeof(SComponentModel));
+IVsPackageUninstaller uninstallerServices =
+    componentModel.GetService<IVsPackageUninstaller>();
+
+var uninstallerServices = installerServices.UninstallPackage(currentProject, "Newtonsoft.json", removeDependencies: true);
+```
+
+## Aditional documentation
```
Comment (jeffkl) [inline src/NuGet.Clients/NuGet.VisualStudio/README.md]:
```suggestion
## Additional documentation
```

**Review by nkolev92 — APPROVED** [review]

**Pull request merged**